### PR TITLE
Added docker guide to docs.md and modified code for upload in server

### PR DESCRIPTION
**Pull request by noobed-max**
now the upload will not replace the files in destination key folder

also fixed the pytest failure over upload endpoint
- #34

### CODE REVIEW

File: docs.md [hunk 1]
```diff
@@ -30,7 +30,97 @@ uvicorn main:app --reload
...
+FROM alpine:latest
+
+# Set the working directory inside the container
+WORKDIR /code
+
+# Copy the requirements file to the working directory
+COPY ./requirements.txt .
+
+# Copy the application files to the working directory
+COPY ./crud.py /code/
```
Comment (Tejas-ChandraShekarRaju) [inline docs.md:54]: Only these two required nothing else? There can be more files in the future. We should be copying the whole directory.

File: server/main.py [hunk 2]
```diff
@@ -35,11 +35,12 @@ async def upload(key: Optional[str] = Form(None), encoded_content: List[str] = F
         key_directory = f"{key}"
```
Comment (Tejas-ChandraShekarRaju) [inline server/main.py:35]: This looks good.

File: docs.md [hunk 3]
```diff
@@ -30,7 +30,97 @@ uvicorn main:app --reload
 Once the app is successfully installed, access http://127.0.0.1:8000/docs, You'll see a screen similar to the below screen shot
 <img width="1617" alt="image" src="https://github.com/cia-labs/Storage-service/assets/41864599/e8774034-5a50-4e82-9e4b-c3e84c47bdf9">
 
+---
+---
 
+# Deploying the Storage Service as a Container
```
Comment (Tejas-ChandraShekarRaju) [inline docs.md:36]: As a Docker Container*

**Review by Tejas-ChandraShekarRaju — CHANGES_REQUESTED** [review]
LGTM. With small review comments.

File: server/main.py [hunk 4]
```diff
@@ -36,11 +36,11 @@ async def upload(key: Optional[str] = Form(None), encoded_content: List[str] = F
         directory_key = f"./storage/{key_directory}"
         os.makedirs(directory_key , exist_ok=True)
         existing_files_count = len([name for name in os.listdir(directory_key) if name.startswith("encodedtxt")])
-        for i, encoded_item in enumerate(separated_strings, start=existing_files_count + 1):
+        for i, encoded_item in enumerate(encoded_content, start=existing_files_count + 1):
```
Comment (Tejas-ChandraShekarRaju) [inline server/main.py:39]: This changes should only be present in update flow and not in upload. If i remember correctly in upload flow we are not allowing to update. If the key exists in store we just return " duplicate key - if you are trying to append please use append functionality"

**Review by Tejas-ChandraShekarRaju — CHANGES_REQUESTED** [review]

File: docs.md (hunk 1 above)
Comment (noobed-max) [reply docs.md:54]: yep for now

Comment (Tejas-ChandraShekarRaju) [reply docs.md:54]: copy the whole directory it should always be generic

File: server/tests/test.py [hunk 5]
```diff
@@ -68,8 +68,8 @@ def test_retrieve_file_key_not_found():
 
 
 def test_update_files_success():
-    key = "test_key"
-    new_key = "new_test_key"
+    key = "test_key1"
```
Comment (noobed-max) [inline server/tests/test.py:71]: assigned different key due to conflicts with upload
as now upload will not override the existing old files

**Pull request closed without merging**